src/plugins/data_source_management/public/components/data_source_menu/types.ts

```diff
@@ -81,6 +82,7 @@ export interface DataSourceSelectableConfig extends DataSourceBaseConfig {
   notifications: NotificationsStart;
   activeOption?: DataSourceOption[];
   dataSourceFilter?: (dataSource: SavedObject<DataSourceAttributes>) => boolean;
+  onChangeDevToolsModalVisible: () => void;
```

Comment:
I thought we will just listen to application change while the modal is visible.

Reply:
Thanks for the comment, the dev tool is just EuiOverlayMask and using core.application.currentAppId$ to monitor app ID only returns the app that opens the dev tool.